### PR TITLE
feat(lambda): migrate stages from AWS Lambda plugin to OSS

### PR DESCRIPTION
**Pull request by mattgogerly**
Moving the Orca component of https://github.com/spinnaker-plugins/aws-lambda-deployment-plugin-spinnaker to OSS.

I've cleaned up a whole load of low hanging fruit code smells in the process (stuff like inlining returns, type inference for collections etc). There's a whole load of other cleanup (constructor injection over field injection as an example) that's still needed.

Note that the plugin is pretty much built around calling Clouddriver directly rather than using existing services like `KatoService`. The bulk of this is in `LambdaCloudDriverUtils`, which creates its own `OkHttpClient` rather than a static one. Because of this the plugin has separate network configuration to the rest of Orca.

In the interests of keeping this migration as safe and as easy for users as possible I'm saving refactoring of this for a future PR.

To follow:
- Deck PR
- Breaking change release note

### CODE REVIEW

File: orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/ClouddriverLambdaConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2018 Netflix, Inc.
```
Comment (dbyron-sf) [inline orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/ClouddriverLambdaConfiguration.java:2]: JPMorgan?

File: orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/LambdaConfigurationProperties.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.config;
+
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Data
+@ConfigurationProperties(prefix = "lambda")
```
Comment (dbyron-sf) [inline orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/LambdaConfigurationProperties.java:23]: FWIW [the plugin](https://github.com/spinnaker-plugins/aws-lambda-deployment-plugin-spinnaker/blob/ae9aca40cfce86d43b0d372939ae80e1a45a82ba/lambda-deployment-orca/src/main/java/com/amazon/aws/spinnaker/plugin/lambda/Config.java#L8) uses `lambdaPluginConfig`.  I think this change is totally fine.

File: orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/lambda/LambdaDeleteStage.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Netflix, Inc.
```
Comment (dbyron-sf) [inline orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/lambda/LambdaDeleteStage.java:2]: The plugin has
```
* Copyright 2018 Amazon.com, Inc. or its affiliates.
 ```

File: orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/lambda/LambdaDeploymentStage.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Netflix, Inc.
```
Comment (dbyron-sf) [inline orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/lambda/LambdaDeploymentStage.java:2]:
```suggestion
 * Copyright 2018 Amazon.com, Inc. or its affiliates.
```

Comment (dbyron-sf) [reply orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/lambda/LambdaDeploymentStage.java:2]: and elsewhere...

**Review by dbyron-sf — APPROVED** [review]

**Comment by dbyron-sf** [comment]
@mergifyio update

**Comment by mattgogerly** [comment]
I'll sort the license out tomorrow.

**Pull request merged**